packages/core/src/electron-main/electron-main-application.ts

```diff
@@ -232,8 +232,11 @@ export class ElectronMainApplication {
             minWidth: 200,
             minHeight: 120,
             webPreferences: {
-                nodeIntegration: true, // https://github.com/eclipse-theia/theia/issues/2018
-                nodeIntegrationInWorker: true
+                // https://github.com/eclipse-theia/theia/issues/2018
+                nodeIntegration: true,
+                // Setting the following option to `true` causes some features to break, somehow.
+                // Issue: https://github.com/eclipse-theia/theia/issues/8577
+                nodeIntegrationInWorker: false,
```

Comment:
@tsmaeder Was anything fixed by enabling this option? It seems to be the origin of the formatting issue on Electron.

Reply:
It is necessary if we want our plugin rpc stuff to work in electron, see https://community.theia-ide.org/t/buffer-not-defined-in-webworker/1031
Since we have no solution for front-end plugins in electron right now anyway, it is not critical, but I think we need a solution for this: either we have the same environment in electron and browser cases or we need to find a way to keep stuff that works out of certain code (i.e. no buffer in plugin rpc).

Reply:
From the issue you linked, it seems to be related to the `Buffer` class missing when running on Electron with the node integration disabled. Maybe something we can fix as a follow up.

Reply:
Opened an issue for this: https://github.com/eclipse-theia/theia/issues/8583